src/app/qgsdockablewidgethelper.h

```diff
@@ -0,0 +1,77 @@
+/***************************************************************************
+  qgsdockablewidgethelper.h
+  --------------------------------------
+  Date                 : January 2022
+  Copyright            : (C) 2022 by Belgacem Nedjima
+  Email                : belgacem dot nedjima at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSDOCKABLEWIDGETHELPER_H
+#define QGSDOCKABLEWIDGETHELPER_H
+
+#include "qgis_app.h"
+#include "qgsdockwidget.h"
+
+#include <QDialog>
+#include <QVBoxLayout>
+
+#define SIP_NO_FILE
+
+class QgsDockWidget;
+
+class APP_EXPORT QgsDockableWidgetHelper : public QWidget
+{
+    Q_OBJECT
+  public:
+    QgsDockableWidgetHelper( bool isDocked, QWidget *widget );
+    ~QgsDockableWidgetHelper();
+
+    void setWidget( QWidget *widget );
+
+    QWidget *widget() { return mWidget; }
+
+    QgsDockWidget *dockWidget() { return mDock; }
+
+    QDialog *dialog() { return mDialog; }
+
+    bool isDocked() const;
```

Comment:
`isDocked` and `isDockFloating` seem to be referring to the same thing, but they are very different... maybe we should rename `isDocked` to something else?

Reply:
They are not referring to the same thing: isDocked indicates whether it a dialog or a dock widget is used to display the widget whereas isDockFloating refers to whether the dock widget is floating on top of other widget or is docked